frontend/catalyst/compilation_pipelines.py

```diff
@@ -449,34 +432,27 @@ class QJIT:
             after the Python process ends. If ``False``, these representations
             will instead be stored in a temporary folder, which will be deleted
             as soon as the QJIT instance is deleted.
+        pipelines (Optional(List[AnyType]): A list of pipelines to be executed. The elements of
+            the list are asked to implement a run method which takes the output of the previous run
+            as an input to the next element, and so on.
         compile_options (Optional[CompileOptions]): Common compilation options
     """
 
-    def __init__(self, fn, target, keep_intermediate, compile_options=None):
+    # pylint: disable=too-many-arguments
+    def __init__(self, fn, target, keep_intermediate, pipelines, compile_options=None):
         self.qfunc = fn
         self.c_sig = None
         functools.update_wrapper(self, fn)
-        if keep_intermediate:
-            dirname = fn.__name__
-            parent_dir = os.getcwd()
-            path = os.path.join(parent_dir, dirname)
-            os.makedirs(path, exist_ok=True)
-            self.workspace_name = path
-        else:
-            # The temporary directory must be referenced by the wrapper class
-            # in order to avoid being garbage collected
-            # pylint: disable=consider-using-with
-            self.workspace = tempfile.TemporaryDirectory()
-            self.workspace_name = self.workspace.name
         self.compile_options = compile_options
-        self.passes = {}
+        self._compiler = Compiler()
+        self.pipelines = pipelines
         self._jaxpr = None
         self._mlir = None
```

Comment:
Not a big deal but I think using a tailing underscore in the name of class attributes is not considered as the recommended naming convention in Python.

Reply:
This is how you denote "private" attributes in Python, or what do you mean?

Reply:
Hmm.. right! it can be a useful convention for "private" attributes. :+1: